drivers/storage/portworx/component/disruption_budget.go

```diff
@@ -124,10 +127,38 @@ func (c *disruptionBudget) createPortworxPodDisruptionBudget(
 		return fmt.Errorf("failed to enumerate nodes: %v", err)
 	}
 
+	k8sNodeList := &v1.NodeList{}
+	err = c.k8sClient.List(context.TODO(), k8sNodeList)
+	if err != nil {
+		return err
+	}
+	k8sNodesStoragePodCouldRun := make(map[string]bool)
+	for _, node := range k8sNodeList.Items {
+		shouldRun, shouldContinueRunning, err := pxutil.CheckPredicates(&node, cluster, nil)
+		if err != nil {
+			return err
+		}
+		if shouldRun || shouldContinueRunning {
+			k8sNodesStoragePodCouldRun[node.Name] = true
+		}
+	}
+
 	storageNodesCount := 0
 	for _, node := range nodeEnumerateResponse.Nodes {
+		if node.SchedulerNodeName == "" {
+			k8sNode, err := coreops.Instance().SearchNodeByAddresses(
+				[]string{node.DataIp, node.MgmtIp, node.Hostname},
+			)
+			if err != nil {
+				logrus.Warnf("Unable to find kubernetes node name for nodeID %v: %v", node.Id, err)
```

Comment:
I don't think this is a good idea because that will spam the log a lot as our default logging is `debug` and we will run this code every 30 seconds. Especially in a large cluster. We can output it as trace log.
Overall we should take a pass over our logging later and keep `info` as the default log.

Reply:
this line should not be executed, it's unexpected if we hit here.

Reply:
never mind then, metro DR setup will not hit here, as SchedulerNodeName will not be empty.

Reply:
I remember PIyush mentioned it's not common we have an empty `SchedulerNodeName` but once we hit that, there will be log spam, so I'll leave it as is. @haibinxie